### DISCUSSION TITLE
Getting errors attempting to run tests.

### DISCUSSION THREAD

**Discussion by githubforw0rk** [ialbert/biostar-central#902] [Q&A] [answered]
I ran:

coverage run manage.py test biostar.forum 

and got an error which I don't think describes the issue at all.  
![image](https://user-images.githubusercontent.com/99474436/163650716-b12afce5-a2a0-48ed-9f9b-ae089c477217.png)

i'm on django 3.2.6 and this runs the application fine on uwsgi and the dev server but for some reason, the test command does not work.

**Comment by ialbert** [comment]
it is a django issue, this type of error usually appears when the DJANGO_SETTING_MODULE is not set properly.

**Reply by githubforw0rk** [reply]
I set the DJANGO_SETTINGS_MODULE to biostar.settings and got the same error as in the screenshot.  I then pulled a fresh copy of code from github and was able to run the tests but had a lot of errors.  After googling some of the errors it looks like sqlite may be having concurrency problems and there were some failures as well:
![image](https://user-images.githubusercontent.com/99474436/163693057-a4905d91-15e7-4bac-844d-75f422a4e1ab.png)

**Comment by githubforw0rk** [comment, marked as answer]
An update:  I was comparing my settings file with the one in the repository and discovered that the missing models issue is reproducible 100% of the time when DEBUG_TOOLBAR is True and DEBUG = False.  When the DEBUG_TOOLBAR is False, I don't get the missing models issue, but Sqlite3 complains about locked databases.  For this issue, I would probably need to install postgres or another database to get around sqlite's limitations.  It does appear that the tests are being run in parallel and sqlite3 cannot handle that many connections at once.  There probably is also a setting to make all requests to sqlite3 be handled synchronously but I'll have to look at that.
